src/include/migraphx/generic_float.hpp

```diff
@@ -0,0 +1,410 @@
+    }
+
+    friend constexpr bool operator!=(const generic_float& x, const generic_float& y)
+    {
+        return not(x == y);
+    }
+
+    constexpr generic_float& operator++() noexcept
+    {
+        *this += generic_float(1.0f);
+        return *this;
+    }
+};
+
+} // namespace MIGRAPHX_INLINE_NS
+} // namespace migraphx
+
+namespace std {
+
+template <unsigned int E, unsigned int M, unsigned int F>
+class numeric_limits<migraphx::generic_float<E, M, F>>
+{
+    public:
+    static constexpr bool has_infinity = true;
+    static constexpr migraphx::generic_float<E, M, F> epsilon()
+    {
+        return migraphx::generic_float<E, M, F>::epsilon();
+    }
+
+    static constexpr migraphx::generic_float<E, M, F> quiet_NaN()
+    {
+        return migraphx::generic_float<E, M, F>::qnan();
+    }
+
+    static constexpr migraphx::generic_float<E, M, F> signaling_NaN()
+    {
+        return migraphx::generic_float<E, M, F>::snan();
+    }
+
+    static constexpr migraphx::generic_float<E, M, F> max()
+    {
+        return migraphx::generic_float<E, M, F>::max();
+    }
+
+    static constexpr migraphx::generic_float<E, M, F> min()
+    {
+        return migraphx::generic_float<E, M, F>::min();
+    }
+
+    static constexpr migraphx::generic_float<E, M, F> lowest()
+    {
+        return migraphx::generic_float<E, M, F>::lowest();
+    }
+
+    static constexpr migraphx::generic_float<E, M, F> infinity()
+    {
+        return migraphx::generic_float<E, M, F>::infinity();
+    }
+
+    static constexpr migraphx::generic_float<E, M, F> denorm_min()
+    {
+        return migraphx::generic_float<E, M, F>::denorm_min();
+    }
+};
+
+template <unsigned int E, unsigned int M, unsigned int F, class T>
+struct common_type<migraphx::generic_float<E, M, F>, T> : std::common_type<float, T> // NOLINT
+{
+};
+
+template <unsigned int E, unsigned int M, unsigned int F, class T>
+struct common_type<T, migraphx::generic_float<E, M, F>> : std::common_type<float, T> // NOLINT
+{
+};
+
+// template<unsigned int E, unsigned int M, unsigned int F, bool FNUZ>
+// struct common_type<migraphx::generic_float<E, M, F>,
+// migraphx::fp8::float8<migraphx::fp8::f8_type::fp8, FNUZ>> : std::common_type<float, float>
+// {};
+
+// template<unsigned int E, unsigned int M, unsigned int F, bool FNUZ>
+// struct common_type<migraphx::fp8::float8<migraphx::fp8::f8_type::fp8, FNUZ>,
+// migraphx::generic_float<E, M, F>> : std::common_type<float, float>
+// {};
+
+// template<unsigned int E, unsigned int M, unsigned int F, migraphx::fp8::f8_type T, bool FNUZ>
+// struct common_type<migraphx::generic_float<E, M, F>, migraphx::fp8::float8<T, FNUZ>> :
+// std::common_type<float, float>
+// {};
+
+// template<unsigned int E, unsigned int M, unsigned int F, migraphx::fp8::f8_type T, bool FNUZ>
+// struct common_type<migraphx::fp8::float8<T, FNUZ>, migraphx::generic_float<E, M, F>> :
+// std::common_type<float, float>
+// {};
```

Comment:
What's the current outlook for the fp8 types? Will we keep the fp8 data structure on ref MIGX or will we specialize generic_float for the fp8 types?